File: Negocio/Basicas/Seguridad.vb
Imports System.Data
Imports System.Diagnostics
Imports System.Threading
Imports System.Security.Cryptography
Imports System.Text

Namespace LiquidacionSueldos
    Namespace Negocio
        Public Class Seguridad
            Shared ocdGestor As New LiquidacionSueldos.Datos.Gestor
            Shared Tabla As New DataTable

            Public Shared Function Autenticar(ByVal usuNombreIngreso As String, ByVal usuClave As String) As DataTable
                Dim ClaveEncriptada As String = EncriptarClave(usuClave)
                Return ocdGestor.EjecutarReader("[Usuario.ObtenerAutenticar]", New Object(,) {{"@usuNombreIngreso", usuNombreIngreso}, {"@usuClave", ClaveEncriptada}})
            End Function

            Public Shared Function ValidarIngresoPagina(ByVal perId As Integer, ByVal PaginaNombre As String) As Boolean
                Tabla = ocdGestor.EjecutarReader("[Usuario.ValidarPagina]", New Object(,) {{"@perId", perId}, {"@PaginaNombre", PaginaNombre}})

                If Tabla.Rows.Count > 0 Then
                    Return True
                Else
                    Return False
                End If
            End Function

            Public Shared Function ObtenerMenu(ByVal menId As Integer, ByVal perId As Integer) As DataTable
                Return ocdGestor.EjecutarReader("[Perfil.ObtenerMenu]", New Object(,) {{"@menId", menId}, {"@perId", perId}})
            End Function

            Public Shared Function EncriptarClave(ByVal usuClave As String) As String
                Dim encoder As UnicodeEncoding = Nothing
                Dim sSHA512 As SHA512Managed = Nothing
                Try
                    encoder = New UnicodeEncoding()
                    sSHA512 = New SHA512Managed()

                Catch oError As Exception
                    Throw oError
                End Try

                Return Convert.ToBase64String(sSHA512.ComputeHash(encoder.GetBytes(usuClave)))
            End Function
        End Class
    End Namespace
End Namespace

File: NegocioCSharp/Basicas/Seguridad.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Threading;
using System.Security.Cryptography;
using System.Text;

namespace LiquidacionSueldos
{
    namespace Negocio
    {
        public class Seguridad
        {
            static Datos.Gestor ocdGestor = new Datos.Gestor();

            static DataTable Tabla = new DataTable();
            public static DataTable Autenticar(string usuNombreIngreso, string usuClave)
            {
                string ClaveEncriptada = EncriptarClave(usuClave);
                return ocdGestor.EjecutarReader("[Usuario.ObtenerAutenticar]", new object[,] {{"@usuNombreIngreso",usuNombreIngreso},{"@usuClave",ClaveEncriptada}});
            }

            public static bool ValidarIngresoPagina(int perId, string PaginaNombre)
            {
                Tabla = ocdGestor.EjecutarReader("[Usuario.ValidarPagina]", new object[,] {
                    {
                        "@perId",
                        perId
                    },
                    {
                        "@PaginaNombre",
                        PaginaNombre
                    }
                });

                if (Tabla.Rows.Count > 0)
                {
                    return true;
                }
                else
                {
                    return false;
                }
            }

            public static DataTable ObtenerMenu(int menId, int perId)
            {
                return ocdGestor.EjecutarReader("[Perfil.ObtenerMenu]", new object[,] {
                    {
                        "@menId",
                        menId
                    },
                    {
                        "@perId",
                        perId
                    }
                });
            }

            public static string EncriptarClave(string usuClave)
            {
                UnicodeEncoding encoder = null;
                SHA512Managed sSHA512 = null;
                try
                {
                    encoder = new UnicodeEncoding();
                    sSHA512 = new SHA512Managed();

                }
                catch (Exception oError)
                {
                    throw oError;
                }

                return Convert.ToBase64String(sSHA512.ComputeHash(encoder.GetBytes(usuClave)));
              
            }
        }
    }
}